src/cport/modules/predictprotein.py

```diff
@@ -0,0 +1,184 @@
+import logging
+import time
+import sys
+import os
+import zipfile
+import glob
+import pandas as pd
+import tempfile
+import shutil
+
+from selenium import webdriver
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.chrome.options import Options
+from selenium.common.exceptions import NoSuchElementException
+
+from cport.url import PREDICTPROTEIN_URL
+from cport.modules import utils
+
+log = logging.getLogger("cportlog")
+
+# Total wait (seconds) = WAIT_INTERVAL * NUM_RETRIES
+WAIT_INTERVAL = 10  # seconds
+NUM_RETRIES = 12
+
+
+class Predictprotein:
+    def __init__(self, pdb_id, chain_id):
+        self.pdb_id = pdb_id
+        self.chain_id = chain_id
+        self.wait = WAIT_INTERVAL
+        self.tries = NUM_RETRIES
+
+    def submit(self):
+        """Makes a submission to the PredictProtein server"""
+        fasta_code = utils.get_fasta_from_pdbid(self.pdb_id, self.chain_id)
+
+        # headless so that browser windows are not visually opened and closed
+        options = Options()
+        options.add_argument("headless")
+
+        # chrome has the most flexibility and ease-of-use
+        driver = webdriver.Chrome(chrome_options=options)
+        driver.get(PREDICTPROTEIN_URL)
+
+        # identifies the textarea
+        elem = driver.find_element_by_id("sequence")
+        elem.clear()
+        elem.send_keys(fasta_code)
+
+        # selects the specific button for submission
+        driver.find_element_by_css_selector(
+            "a.btn.btn-submit.btn-primary.btn-large"
+        ).click()
+
+        # sleep so that the page is properly loaded before continuing
+        time.sleep(5)
+
+        # if the submission occurred before the result page will be presented
+        html = driver.current_url
+
+        # if the submission has never occurred before a link will be shown to await results
+        try:
```

Comment:
After some digging, it appears that by using a different function this can indeed be avoided. It now uses a list which is either empty if the link is not present, or contains the element with the link if it is present. Now it just has to check if that list is empty or not.

Reply:
nice!